pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -128,11 +130,14 @@ public void init(PinotConfiguration config, InstanceDataManager instanceDataMana
     try {
       long releaseMs = config.getProperty(QueryConfig.KEY_OF_SCHEDULER_RELEASE_TIMEOUT_MS,
           QueryConfig.DEFAULT_SCHEDULER_RELEASE_TIMEOUT_MS);
-      _queryWorkerExecutorService = Executors.newFixedThreadPool(ResourceManager.DEFAULT_QUERY_WORKER_THREADS,
+      _queryWorkerIntermediateExecutorService = Executors.newFixedThreadPool(
```

Comment:
This should be `newCachedThreadPool`?

Reply:
We can limit the number of threads b/c when a opChain is waiting on mailbox it will back out from the worker pool and queue as unsubmitted opChain, when wake up by mailbox it will be re-added back to the worker pool